test/common/upstream/health_checker_impl_test.cc

```diff
@@ -2467,6 +2467,121 @@ TEST_F(HttpHealthCheckerImplTest, Http2ClusterUseHttp2CodecClient) {
   EXPECT_EQ(Http::CodecClient::Type::HTTP2, health_checker_->codecClientType());
 }
 
+MATCHER_P(MetadataEq, expected, "") {
+  const envoy::config::core::v3::Metadata* metadata = arg;
+  if (!metadata) {
+    return false;
+  }
+  EXPECT_TRUE(Envoy::Protobuf::util::MessageDifferencer::Equals(*metadata, expected));
+  return true;
+}
+
+TEST_F(HttpHealthCheckerImplTest, TransportSocketMatchCriteria) {
+  const std::string host = "fake_cluster";
+  const std::string path = "/healthcheck";
+  const std::string yaml = R"EOF(
+    timeout: 1s
+    interval: 1s
+    no_traffic_interval: 1s
+    interval_jitter_percent: 40
+    unhealthy_threshold: 2
+    healthy_threshold: 2
+    http_health_check:
+      service_name_matcher:
+        prefix: locations
+      path: /healthcheck
+    transport_socket_match_criteria:
+      key: value
+    )EOF";
+
+  auto default_socket_factory = std::make_unique<NiceMock<Network::MockTransportSocketFactory>>();
+  // We expect that this default_socket_factory will NOT be used to create a transport socket for
+  // the health check connection.
+  EXPECT_CALL(*default_socket_factory, createTransportSocket(_)).Times(0);
```

Comment:
are there other calls to be made against this mock? if so you could forego this and just make the mock a strict mock (every call needs an expectation) by not making it a `NiceMock`

Reply:
The socket factory also calls `implementsSecureTransport`. It isn't impacted by this test case so I left it out, but I can add it back into be explicit.

Reply:
Oh I missed what you were trying to say.  I forgot this is where I was calling `.Times(0)`. 

Since this is very much behavior that this case is testing, I like being explicit. What do you think?

Reply:
You can be explicit about a `.Times(0)` with a strict mock too, it just wont actually do anything